added get_program by id on ledger get.rs

FranFiuba opened the pull request:
## Motivation
We need a get program by id function, so we can create an endpoint on the snark/beacon repository.

FranFiuba committed:
added get_program by id on ledger get.rs

FranFiuba committed:
getprogram now recives a programid as a string

raychu86 commented on vm/compiler/src/ledger/get.rs:
```diff
@@ -84,6 +84,15 @@ impl<N: Network, B: BlockStorage<N>, P: ProgramStorage<N>> Ledger<N, B, P> {
         }
     }
 
+    // Returns the program for the given program_id.
+    pub fn get_program(&self, program_id: String) -> Result<Program<N>> {
+        let program_id = ProgramID::<N>::from_str(&program_id)?;
```
```suggestion
    // Returns the program for the given program ID.
    pub fn get_program(&self, program_id: ProgramID<N>) -> Result<Program<N>> {
```

FranFiuba committed:
change input to u64

FranFiuba committed:
change input to u32

FranFiuba committed:
fix a mistake: program_id is a string

Commit:
Update suggested changes

- Expected program_id should be ProgramID instead of String
- Function comment should be a documentation comment instead of a regular one
